controllers/authorino_controller.go

```diff
@@ -424,9 +429,39 @@ func (r *AuthorinoReconciler) authorinoDeploymentChanges(existingDeployment, des
 	return false
 }
 
+func authorinoServiceAccountName(crName string) string {
+	return fmt.Sprintf("%s-authorino", crName)
+}
+
+func (r *AuthorinoReconciler) createAuthorinoServiceAccount(authorino *api.Authorino) (*k8score.ServiceAccount, error) {
+	var logger = r.Log
+	saName := authorinoServiceAccountName(authorino.Name)
+	saNamespace := authorino.Namespace
+	sa := &k8score.ServiceAccount{}
+	if err := r.Get(context.TODO(), namespacedName(saNamespace, saName), sa); err != nil {
+		if errors.IsNotFound(err) {
+			// ServiceAccount doesn't exit - create one
+			sa.Name = saName
+			sa.Namespace = saNamespace
+			_ = ctrl.SetControllerReference(authorino, sa, r.Scheme)
+			if err := r.Client.Create(context.TODO(), sa); err != nil {
+				return nil, r.wrapErrorWithStatusUpdate(
+					logger, authorino, r.setStatusFailed(api.AuthorinoUnableToCreateServiceAccount),
+					fmt.Errorf("failed to create %s ServiceAccount, err: %v", saName, err),
+				)
+			}
+		}
+		return nil, r.wrapErrorWithStatusUpdate(
+			logger, authorino, r.setStatusFailed(api.AuthorinoUnableToGetServiceAccount),
+			fmt.Errorf("failed to get %s ServiceAccount, err: %v", saName, err),
+		)
+	} else {
```

Comment:
Do we need this else statement here? We could only return the service account because if an error occur it will be returned already as you can see in line 454

Reply:
Line 454 returns an error if the getting the service account fails to a reason other than a 404 (quite unusual). This one is for when the service account already exists (i.e., getting the service account did not fail), and therefore the process can move on.

Reply:
Not sure if I am missing something but I didn't understand the reason for adding the else statement, because if getting the services account doesn't fail then the method will return the retrieved service account anyway

Reply:
What do you mean by "anyway"? The `else` statement _is_ where we return the SA when getting the SA does not fail.